packages/sveltekit/test/client/vendor/lookUpCache.test.ts

```diff
@@ -0,0 +1,68 @@
+import * as utils from '@sentry/utils';
+import { vi } from 'vitest';
+
+import { isRequestCached } from '../../../src/client/vendor/lookUpCache';
+
+let scriptElement: {
+  textContent: string;
+  getAttribute: (name: string) => string | null;
+} | null;
+
+vi.spyOn(utils, 'getDomElement').mockImplementation(() => {
+  return scriptElement;
+});
+
+describe('isRequestCached', () => {
+  it('should return true if a script tag with the same selector as the constructed request selector is found', () => {
+    scriptElement = {
+      textContent: 'test',
+      getAttribute: () => null,
+    };
+
+    expect(isRequestCached('/api/todos/1', undefined)).toBe(true);
+  });
+
+  it('should return false if a script with the same selector as the constructed request selector is not found', () => {
+    scriptElement = null;
+
+    expect(isRequestCached('/api/todos/1', undefined)).toBe(false);
+  });
+
+  it('should return true if a script with the same selector as the constructed request selector is found and its TTL is valid', () => {
+    scriptElement = {
+      textContent: 'test',
+      getAttribute: () => (performance.now() / 1000 + 1).toString(),
```

Comment:
Wondering if this might ever be flakey... I think it's very unlikely but maybe we wanna use something like https://stackoverflow.com/a/63377110/21967320 just to show best practice? I will leave this up to you!

Reply:
Not sure about flakiness but useFakeTimers sounds better either way. [3840964](https://github.com/getsentry/sentry-javascript/pull/8391/commits/384096486091670f3c415db5cf85d6bae3fea024)